dd-java-agent/src/main/java/datadog/trace/bootstrap/BootstrapInitializationTelemetry.java

```diff
@@ -126,7 +130,32 @@ public void onFatalError(Throwable t) {
 
     @Override
     public void onError(String reasonCode) {
+      error = true;
       onPoint("library_entrypoint.error", "error_type:" + reasonCode);
+      setMetadata("error", mapResultClass(reasonCode), reasonCode);
+    }
+
+    private void setMetadata(String result, String resultClass, String resultReason) {
```

Comment:
I would rename the result helper method and use overloading for clarity. Something like:

```suggestion
    private void setMetaResult(String result, String resultCode) {
      setMetaResult(result,  mapResultClass(reasonCode), reasonCode);
    }
    
    private void setMetaResult(String result, String resultClass, String resultReason) {
    
    }
```

WDYT?